core/services/eth/pool.go

```diff
@@ -5,83 +5,129 @@ import (
+		logger.Named("Pool").With("evmChainID", chainID.String()),
+		make(chan struct{}),
+		sync.WaitGroup{},
 	}
 	return p
 }
 
-func (p *Pool) initChainID(chainID *big.Int) {
-	p.chainID = chainID
-	p.logger = p.logger.With("evmChainID", chainID.String())
-}
-
 // Dial dials every node in the pool and verifies their chain IDs are consistent.
-func (p *Pool) Dial(ctx context.Context) (err error) {
-	for _, n := range p.nodes {
-		err = multierr.Combine(err, n.Dial(ctx))
-	}
-	for _, s := range p.sendonlys {
-		err = multierr.Combine(err, s.Dial(ctx))
-	}
-	if err != nil {
-		return err
-	}
-	return p.verifyChainIDs(ctx)
+func (p *Pool) Dial(ctx context.Context) error {
+	return p.StartOnce("Pool", func() (merr error) {
+		for _, n := range p.nodes {
+			if err := n.Dial(ctx); err != nil {
+				p.logger.Errorw("Error dialing node", "node", n, "err", err)
+			} else if err := n.Verify(ctx, p.chainID); err != nil {
+				p.logger.Errorw("Error verifying node", "node", n, "err", err)
+			}
+		}
+		for _, s := range p.sendonlys {
+			// TODO: Deal with sendonly nodes state
+			err := s.Dial(ctx)
+			if err != nil {
+				return err
+			}
+		}
+		p.wg.Add(1)
+		go p.runLoop()
+
+		return nil
+	})
 }
 
-// verifyChainIDs checks that every node's chain ID is consistent, initializing from the first node if nil.
-func (p *Pool) verifyChainIDs(ctx context.Context) (err error) {
-	if p.chainID == nil {
-		chainID, err2 := p.nodes[0].ChainID(ctx)
-		if err2 != nil {
-			return errors.Wrap(err, "failed to get chain ID from first node")
+// DialRetryInterval controls how often we try to reconnect a dead node
+var DialRetryInterval = 5 * time.Second
+
+func (p *Pool) runLoop() {
+	defer p.wg.Done()
+	ticker := time.NewTicker(DialRetryInterval)
+
+	for {
+		select {
+		case <-p.chStop:
+			return
+		case <-ticker.C:
+			// re-dial all dead nodes
+			func() {
+				ctx, cancel := utils.ContextFromChan(p.chStop)
+				defer cancel()
+				ctx, cancel = context.WithTimeout(ctx, DialRetryInterval)
+				defer cancel()
+				// TODO: How does this play with automatic WS reconnects?
+				p.redialDeadNodes(ctx)
+			}()
 		}
-		p.initChainID(chainID)
 	}
+}
+
+func (p *Pool) redialDeadNodes(ctx context.Context) {
 	for _, n := range p.nodes {
-		err = multierr.Combine(err, n.Verify(ctx, p.chainID))
-	}
-	for _, s := range p.sendonlys {
-		err = multierr.Combine(err, s.Verify(ctx, p.chainID))
+		if n.State() == NodeStateDead {
+			if err := n.Dial(ctx); err != nil {
+				p.logger.Errorw(fmt.Sprintf("Failed to redial eth node: %v", err), "err", err)
+			}
+		}
+		if n.State() == NodeStateInvalidChainID {
+			if err := n.Verify(ctx, p.chainID); err != nil {
+				p.logger.Errorw(fmt.Sprintf("Failed to verify eth node: %v", err), "err", err)
+			}
+		}
 	}
-	return err
 }
 
 func (p *Pool) Close() {
-	for _, n := range p.nodes {
-		n.Close()
+	// have to handle err here to avoid linter even though it will always be
+	// nil
+	err := p.StopOnce("Pool", func() error {
```

Comment:
Does it accept explicit suppression?
```suggestion
	_ = p.StopOnce("Pool", func() error {
```

Reply:
You could use `//nolint:errcheck` here